packages/core/admin/server/policies/hasLicenseSeatsLimitAllowance.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const utils = require('@strapi/utils');
+
+const { PolicyError } = utils.errors;
+
+module.exports = async (policyCtx, config = {}) => {
+  if (!strapi.EE) {
+    return true;
+  }
+
+  const permittedSeats = strapi.ee.licenseInfo.seats;
+  if (!permittedSeats) {
+    return true;
+  }
+
+  const userCount = await strapi.service('admin::user').getCurrentActiveUserCount();
+
+  if (userCount < permittedSeats) {
+    return true;
+  }
+
+  if (userCount >= permittedSeats && config.isCreating) {
+    throw new PolicyError("License seat limit reached, can't create new user", {
+      policy: 'license-limit-allowance',
+    });
+  }
+
+  const user = await strapi.service('admin::user').findOne(policyCtx.params.id);
+
+  if (!user) {
+    return true; // Delegate not found to the controller
+  }
+
+  if (
+    userCount >= permittedSeats &&
+    policyCtx.request.body.isActive === true &&
+    user.isActive !== policyCtx.request.body.isActive
+  ) {
+    throw new PolicyError("License seat limit reached, can't reactivate user", {
+      policy: 'license-limit-allowance',
+    });
+  }
```

Comment:
Why checking if seat limit is reached should be a policy?

My personal view:
- This is adding EE logic outside the EE folder.
- It feels like bussiness logic, something that belongs to EE Seats Eforcement, not to a generic admin policy.

I personally think this is logic that should be in a service, when trying to create or update a user, check if the permitted seats surpass the allowed amount. wdyt?

Reply:
Policies are kinda made for that though, no? Otherwise I would need to copy pretty much all controller code to the ee folder for it to work correctly and this would increase code duplication. Happy to look at it together tomorrow and see if you have a better solution 👍

Reply:
you can actually extend a controller using strapi.get('<controller-name').extend. Let me look  for an example.

Reply:
https://github.com/strapi/strapi/blob/main/packages/plugins/i18n/server/services/core-api.js#L186

In i18n we extend all content type controllers to add localization stuff. Maybe we could do the same here?